src/qt3utils/datagenerators/princeton.py

```diff
@@ -0,0 +1,339 @@
+
+        result = np.empty(size)
+
+        for i in range(size):
+            result[i] = self.light.experiment.SystemColumnCalibration[i]
+
+        return result
+
+    @center_wavelength.setter
+    def center_wavelength(self, nanometers):
+        """
+        Sets the spectrometer center wavelength to nanometers.
+        """
+        # this avoids bug where if step and glue is selected, doesn't allow setting center wavelength
+        self.light.set(lf.AddIns.ExperimentSettings.StepAndGlueEnabled, False)
+
+        return self.light.set(lf.AddIns.SpectrometerSettings.GratingCenterWavelength, nanometers)
+
+    @property
+    def grating(self):
+        """
+        Returns the current grating.
+        """
+        return self.light.get(lf.AddIns.SpectrometerSettings.GratingSelected)
+
+    @grating.setter
+    def grating(self, grating):
+        """
+        Sets the current grating to be the one specified by parameter grating.
+        """
+        # TODO: figure out the format for setting this
+
+        print('still need to figure out the format for this')
+
+    @property
+    def gratings(self):
+        """
+        Returns a list of all installed gratings.
+        """
+        break_down = False
+
+        if break_down:
+
+            import re
+
+            for g in GRATINGS:
+                match = re.search(r"\[(\d+\.?\d+[nu]m),(\d+)\]\[(\d+)\]\[(\d+)\])", g)
+                blaze, g_per_mm, slot, turret = match.groups()
+
+        return GRATINGS
+
+    @property
+    def num_frames(self):
+        """
+        Returns the number of frames taken during the acquisition.
+        """
+        return self.light.get(lf.AddIns.ExperimentSettings.FrameSettingsFramesToStore)
+
+    @num_frames.setter
+    def num_frames(self, num_frames):
+        """
+        Sets the number of frames to be taken during acquisition to number.
+        """
+        return self.light.set(lf.AddIns.ExperimentSettings.FrameSettingsFramesToStore, num_frames)
+
+    @property
+    def exposure_time(self):
+        """
+        Returns the single frame exposure time (in ms).
+        """
+        return self.light.get(lf.AddIns.CameraSettings.ShutterTimingExposureTime)
+
+    @exposure_time.setter
+    def exposure_time(self, ms):
+        """
+        Sets the single frame exposure time to be ms (in ms).
+        """
+        return self.light.set(lf.AddIns.CameraSettings.ShutterTimingExposureTime, ms)
+
+    @property
+    def sensor_temperature(self):
+        """
+        Returns the current sensor temperature (in celcius).
+        """
+        return self.light.get(lf.AddIns.CameraSettings.SensorTemperatureReading)
+
+    @property
+    def sensor_setpoint(self):
+        """
+        Returns the sensor setpoint temperature (celcius).
+        """
+        return self.light.get(lf.AddIns.CameraSettings.SensorTemperatureSetPoint)
+
+    @sensor_setpoint.setter
+    def sensor_setpoint(self, deg_C):
```

Comment:
Can we change the name of this?  Seems like 'temperature_sensor_setpoint' or 'sensor_temperature_setpoint' seems more appropriate. 'sensor_setpoint' is too ambiguous.

Also, can you improve the docstring here. What does setting the "setpoint" do here? Is this a calibration? 

If it's too complicated to explain, maybe include a link to the documentation online.

Reply:
Will do. Setpoint is meant to control the temperature setpoint setting, the user inputs the value (most likely (-70 degrees celsius) then the program reads it and calibrates the temperature to this.